aten/src/ATen/native/Linear.cpp

```diff
@@ -646,13 +646,14 @@ Tensor &tensordot_out(const Tensor& input1, const Tensor& input2, IntArrayRef di
   auto result_dtype = result_tmp.scalar_type();
   auto output_tensor_dtype = result.scalar_type();
   auto output_device = result.device();
-  auto input_device = input1.device();
+  auto input1_device = input1.device();
+  auto input2_device = input2.device();
   // check if the input & output tensors are on the same device.
   TORCH_CHECK(
-    output_device == input_device,
+    (output_device == input1_device) && (input1_device == input2_device),
     "tensordot: Expected the output and input tensors to be on the "
-    "same device, but got output on ", output_device, " and inputs on ",
-    input_device);
+    "same device, but got output on ", output_device, ", input1 on ",
+    input1_device, ", and input2 on ", input2_device);
```

Comment:
Hello @heitorschueroff, in the tensordot PR a few days back, I should've checked the device type of both the input tensors, and not just the first one.

Reply:
Sounds good; thanks for noticing

Reply:
The code here is a little confusing because the C++ calls the inputs `input1` and `input2`, but the user-facing names from the documentation are `a` and `b`, and error messages should reflect the user's experience and not the code; so this is a great error message but it needs a slight tweak to replace `input1` with `a` and `input2` with `b`

Reply:
It seems that even [`checkSameDevice`](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/LinearAlgebraUtils.h#L485-L491) in `LinearAlgebraUtils.h` is also a bit lacking in terms of UX.
It only accepts one input, so if `checkSameDevice` is called multiple times for multiple input tensors, then it may not be immediately clear to a user as to which input is not on the same device as the output tensor. So, if you'd like, I can use a `TORCH_CHECK` for `dot` & `vdot` in this PR instead.

`checkAllSameGPU` in `TensorUtils.h` is better in the sense that it lets one provide individual names for tensors, which are included in error messages. But as the name suggests, it only checks if any given number of tensors are on the same GPU.

I proposed adding `checkAllSameDevice` in PR #56628 to check if multiple tensors are on the same device, but then a simple `TORCH_CHECK` might be more appropriate.

Reply:
Using simpler existing utilities, like TORCH_CHECK, sounds like a good solution for now

I agree that eventually we should review and see if there's a simpler, more universal solution